plugins/collector/api/collector.go

```diff
@@ -24,36 +24,22 @@ import (
 	"github.com/apache/skywalking-satellite/internal/pkg/plugin"
 )
 
-//   Init()     Initial stage: Init plugin by config
-//    ||
-//    \/
-//   Prepare()   Preparing stage: Prepare the collector, such as build connection with SkyWalking javaagent.
-//    ||
-//    \/
-//   Next()     Running stage: When Collector collect a data, the data would be fetched by the upstream
-//    ||                       component through this method.
-//    \/
-//   Close()    Closing stage: Close the Collector, such as close connection with SkyWalking javaagent.
-
 // Collector is a plugin interface, that defines new collectors.
 type Collector interface {
 	plugin.Plugin
-
-	// Prepare creates a listen or reader to gather data.
-	Prepare()
+	// Prepare creates a listener or reader to gather APM data, such as start a gRPC listener for Segment receiving.
+	Prepare() error
```

Comment:
If we need to implement a `http-receiver`, should we provide concrete implementation here?

Reply:
And do we need a type field to identify the `receiver` protocol, such as `http` `grpc`.

Reply:
I don't think the upper layers need to care about protocols, they're just different plug-ins

Reply:
Take a look again. Related concepts have been added